### ISSUE TITLE
Generated ActionRequest fails with NullPointerException if we provide null for non-required parameters.

### ISSUE THREAD

**Issue by antonantonovich** [davidmoten/odata-client#423]
I use maven plugin to generate client sources. In generated classes I have something like that:
``` 
    @Action(name = "MyCustomAction")
    @JsonIgnore
    public ActionRequestReturningNonCollectionUnwrapped<ComboIdentifier> myCustomAction(String param1, String param2, String param3, String param4) {
        Preconditions.checkNotNull(param1, "param1 cannot be null");
        Preconditions.checkNotNull(param2, "param2 cannot be null");
        Map<String, TypedObject> _parameters = ParameterMap
            .put("param1", "Edm.String", param1)
            .put("param2", "Edm.String", param2)
            .put("param3", "Edm.String", param3)
            .put("param4", "Edm.String", param4)
            .build();
        return new ActionRequestReturningNonCollectionUnwrapped<ComboIdentifier>(this.contextPath.addActionOrFunctionSegment("Domain.MyCustomAction"), ComboIdentifier.class, _parameters);
    }
```

Required parameters verified to be not null. Non required parameters passed as is to the map.
When I execute action like:
```
myCustomAction("param1Value", "param2Value", null, null).get()
```
I will have NPE. It failed in `ActionRequestReturningNonCollection.get` at line 23 where `ParameterMap.toMap(parameters)` called. ParameterMap.toMap method do not support maps with null values.

**Comment by davidmoten** [comment]
Just saw this, sorry. I imagine fix is easy. Will do.

**Comment by davidmoten** [comment]
This appears to be related to a known JDK [bug](https://bugs.openjdk.org/browse/JDK-8148463). Using JDK 21 I couldn't get this test to fail:

```java
    @Test
    public void testNull() {
        Map<String, TypedObject> map = ParameterMap
                .put("param1", "Edm.String", "blah")
                .put("param2", "Edm.String", "hi")
                .put("param3", "Edm.String", null)
                .put("param4", "Edm.String", null)
                .build();
        ParameterMap.toMap(map);
    }
```
I'll do it without `Collectors.toMap` so that older versions of JDK are still supported.

**Comment by davidmoten** [comment]
Fixed in #431

**Comment by davidmoten** [comment]
Fix in [0.1.93](https://github.com/davidmoten/odata-client/releases/tag/0.1.93) on Maven Central now
